src/main/java/org/omnifaces/util/JNDIObjectLocator.java

```diff
@@ -360,7 +223,221 @@ private InitialContext createInitialContext() {
+		 * @throws IllegalStateException When namespace is already set in this builder.
+		 */
+		public JNDIObjectLocatorBuilder global() {
+			return namespace(JNDI_NAMESPACE_GLOBAL);
+		}
+
+		/**
+		 * Specifies that the default namespace to be used in construction of portable JNDI names must be <code>java:app</code> instead of <code>java:module</code>.
+		 * @return This builder.
+		 * @throws IllegalStateException When namespace is already set in this builder.
+		 */
+		public JNDIObjectLocatorBuilder app() {
+			return namespace(JNDI_NAMESPACE_APPLICATION);
+		}
+
+		/**
+		 * Specifies to disable cache. The default is <code>false</code>.
+		 * @return This builder.
+		 * @throws IllegalStateException When noCaching is already set in this builder or when this builder is already build.
+		 */
+                public JNDIObjectLocatorBuilder noCaching() {
+                        this.noCaching = true;
+                        return this;
+                }
+
+		/**
+		 * Specifies to cache remote enterprise beans. The default is <code>false</code>.
+		 * @return This builder.
+		 * @throws IllegalStateException When cacheRemote is already set in this builder or when this builder is already build.
+		 */
+                public JNDIObjectLocatorBuilder cacheRemote() {
+                        this.cacheRemote = true;
+                        return this;
+                }
+
+                /**
+		 * Builds the {@link JNDIObjectLocator}.
+		 * @return The {@link JNDIObjectLocator}.
+		 * @throws IllegalStateException When this builder is already build.
+		 */
+                public JNDIObjectLocator build() {
+                        Map<String, String> environment;
+                        switch (this.environmentKey == null ? 0 : this.environmentKey.size()) {
+                                case 0:
+                                        environment = Collections.emptyMap();
+                                        break;
+                                case 1:
+                                        environment = Collections.singletonMap(this.environmentKey.get(0), this.environmentValue.get(0));
+                                        break;
+                                default:
+                                        environment = new LinkedHashMap<>(this.environmentKey.size() < 1073741824 ? 1 + this.environmentKey.size() +
+                                                (this.environmentKey.size() - 3) / 3 : Integer.MAX_VALUE);
+                                        for (int ii = 0; ii < this.environmentKey.size(); ii++) {
+                                                environment.put(this.environmentKey.get(ii), this.environmentValue.get(ii));
+                                        }
+                                        environment = unmodifiableMap(environment);
+                        }
+                        String namespaceValue = this.namespaceValue;
+                        if (!this.namespaceSet) {
```

Comment:
Why not `if (namespaceValue == null)`?

Reply:
because it's never null, there is always a default (global)